pkg/exchange/bybit/types.go

```diff
@@ -210,3 +212,153 @@ func (k *KLine) toGlobalKLine(symbol string) (types.KLine, error) {
+	FeeRate fixedpoint.Value `json:"feeRate"`
+	// The remaining qty not executed. Normal spot is not supported
+	LeavesQty fixedpoint.Value `json:"leavesQty"`
+	// Order price. Normal spot is not supported
+	OrderPrice fixedpoint.Value `json:"orderPrice"`
+	// Order qty. Normal spot is not supported
+	OrderQty fixedpoint.Value `json:"orderQty"`
+	// Stop order type. If the order is not stop order, any type is not returned. Normal spot is not supported
+	StopOrderType string `json:"stopOrderType"`
+	// Whether to borrow. Unified spot only. 0: false, 1: true. . Normal spot is not supported, always 0
+	IsLeverage string `json:"isLeverage"`
+	// Implied volatility of mark price. Valid for option
+	MarkIv string `json:"markIv"`
+	// The mark price of the symbol when executing. Valid for option
+	MarkPrice fixedpoint.Value `json:"markPrice"`
+	// The index price of the symbol when executing. Valid for option
+	IndexPrice fixedpoint.Value `json:"indexPrice"`
+	// The underlying price of the symbol when executing. Valid for option
+	UnderlyingPrice fixedpoint.Value `json:"underlyingPrice"`
+	// Implied volatility. Valid for option
+	TradeIv string `json:"tradeIv"`
+}
+
+func (t *TradeEvent) toGlobalTrade(symbolFee symbolFeeDetail) (*types.Trade, error) {
+	if t.Category != bybitapi.CategorySpot {
+		return nil, fmt.Errorf("unexected category: %s", t.Category)
+	}
+
+	side, err := toGlobalSideType(t.Side)
+	if err != nil {
+		return nil, err
+	}
+
+	orderIdNum, err := strconv.ParseUint(t.OrderId, 10, 64)
+	if err != nil {
+		return nil, fmt.Errorf("unexpected order id: %s, err: %w", t.OrderId, err)
+	}
+
+	execIdNum, err := strconv.ParseUint(t.ExecId, 10, 64)
+	if err != nil {
+		return nil, fmt.Errorf("unexpected exec id: %s, err: %w", t.ExecId, err)
+	}
+
+	trade := &types.Trade{
+		ID:            execIdNum,
+		OrderID:       orderIdNum,
+		Exchange:      types.ExchangeBybit,
+		Price:         t.ExecPrice,
+		Quantity:      t.ExecQty,
+		QuoteQuantity: t.ExecPrice.Mul(t.ExecQty),
+		Symbol:        t.Symbol,
+		Side:          side,
+		IsBuyer:       side == types.SideTypeBuy,
+		IsMaker:       t.IsMaker,
+		Time:          types.Time(t.ExecTime),
+		Fee:           fixedpoint.Zero,
+		FeeCurrency:   "",
+	}
+	trade.FeeCurrency, trade.Fee = t.CalculateFee(symbolFee)
+	return trade, nil
+}
+
+// CalculateFee given isMaker to get the fee currency and fee.
+// https://bybit-exchange.github.io/docs/v5/enum#spot-fee-currency-instruction
+//
+// with the example of BTCUSDT:
+//
+// Is makerFeeRate positive?
+//
+//   - TRUE
+//     Side = Buy -> base currency (BTC)
+//     Side = Sell -> quote currency (USDT)
+//
+//   - FALSE
+//     IsMakerOrder = TRUE
+//     -> Side = Buy -> quote currency (USDT)
+//     -> Side = Sell -> base currency (BTC)
+//
+//     IsMakerOrder = FALSE
+//     -> Side = Buy -> base currency (BTC)
+//     -> Side = Sell -> quote currency (USDT)
+func (t *TradeEvent) CalculateFee(feeDetail symbolFeeDetail) (string, fixedpoint.Value) {
```

Comment:
Do you think we can use the feeModeFunctionNative function?

Reply:
Oh seems like feeModeFunctionNative is for types.Order, not for trade

Reply:
That's a great idea! Perhaps we can extract price and quantity into interface functions. Then, both the order and trade can implement them. Ultimately, we can transition the order to use the aforementioned interface.

Reply:
We can do it in next PR